### PR TITLE
Add unit testing to ocean_merge program

### LINKED ISSUE #944: Create unit tests for ocean_merge code
Follow the procedures used in other programs:

- Place the 'guts' of the program into Fortran modules that are accessed by a simple driver.
- Modify build to create a library that is linked by the driver.
- Create unit tests that link to the library.

Unit tests should be created for all major functions of the code:

- The read of the orography file.
- The read of the namelist file.
- The loop that merges the land and ocean masks along coastlines.
- The write of the updated orography file.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
Refactor the `ocean_merge` code into subroutines. Add unit testing.

## TESTS CONDUCTED: 
If there are changes to the build or source code, the tests below must be conducted. Contact a repository manager if you need assistance.

- [x] Compile branch on all Tier 1 machines using Intel (Orion, Jet, Hera, Hercules and WCOSS2).
- [x] Compile branch on Hera using GNU. Done using aff1a41.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done with no warnings using aff1a41.
- [x] Run unit tests locally on any Tier 1 machine. All tests passed with aff1a41. (Used IntelLLVM on Hercules and Hera; used Gnu on Hera; used Intel Classic on WCOSS2)
- [x] Run `grid_gen` consistency tests locally on all Tier 1 machines. Done using aff1a41. All passed on Jet, Hera, Hercules, Orion and WCOSS2.

The current `grid_gen` consistency tests don't exercise the `ocean_merge` program, so a C192 grid was created using on Hercules using the script in ./driver_scripts.  Output orography files from the branch at aff1a41 and the develop at 122010e were identical. It was then decided to update one of the tests to invoke `ocean_merge`. This was done at afb59a4. See [comment](https://github.com/ufs-community/UFS_UTILS/pull/997#issuecomment-2486721694)

## DEPENDENCIES:
None.

## DOCUMENTATION:
The new `ocean_merge` routines need Doxygen comments. 

- [x] Create Doxygen with no warnings due to the `ocean_merge` code. Done on Hera using aff1a41.

## ISSUE: 
Fixes #944.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
After some thought, I decided to update the `c96.uniform` regression test to invoke the `ocean_merge` step. Done at afb59a4. The updated test will require a new baseline on all machines.

File: tests/ocean_merge/data/input.nml
Comment (sanatcumar) [inline tests/ocean_merge/data/input.nml:1]: Should this be mx500? as per reg_tests/grid_gen/c96.uniform.sh Line 20 "ocn = 500"?

File: tests/ocean_merge/data/input.nml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+     &mask_nml
+     ocean_mask_dir="/ocean/mask/dir"
+     ocnres="mx025"
```
Comment (sanatcumar) [inline tests/ocean_merge/data/input.nml]: Should this be mx500 as per reg_tests/grid_gen/c96.uniform.sh LIne 20 "ocn = 500"?

Comment (GeorgeGayno-NOAA) [reply tests/ocean_merge/data/input.nml]: Done at 6364736.

**Review by sanatcumar — APPROVED** [review]
Build the latest commit , Ran the default C96.mx100. Identical files as compared to previous runs. Ran the new test on Hera and all tests passed with the updated baselines

**Pull request merged**